var/spack/repos/builtin/packages/npb/package.py

```diff
@@ -66,20 +76,39 @@ class Npb(MakefilePackage):
     )
 
     # TODO: Combine these into a single mutually exclusive variant
-    variant('mpi',    default=True,  description='Build the MPI implementation')
-    variant('openmp', default=False, description='Build the OpenMP implementation')
-    variant('serial', default=False, description='Build the serial version')
-
-    # TODO: Convert these to non-exclusive multi-valued variants
-    variant('names', default=','.join(valid_names),
-            description='Benchmark names (comma separated list)')
-    variant('classes', default=','.join(valid_classes),
-            description='Benchmark classes (comma separated list)')
+    variant(
+        'implementation',
+        default='mpi',
+        values=('serial', 'mpi', 'openmp'),
+        description='Selects one among the available implementations'
+    )
+
+    variant(
+        'names',
```

Comment:
I actually wrote this package, so all renaming arguments can be directed towards me. I still stand by my original names, although I don't really care that much. I agree with @hartzell, let's rename variants in a different PR if we decide to do that.